src/v/cloud_storage/remote_partition.cc

```diff
@@ -184,10 +184,11 @@ class partition_record_batch_reader_impl final
                 }
                 co_return storage_t{std::move(d)};
             }
-        } catch (const ss::gate_closed_exception&) {
+        } catch (const std::exception& e) {
             vlog(
               _ctxlog.debug,
-              "gate_closed_exception while reading from remote_partition");
+              "exception thrown while reading from remote_partition: {}",
```

Comment:
I think this should be two separate exception handlers, one for gate_closed and one for everything else.

For exceptions in general (i.e. those which aren't a harmless thing like gate_closed), we should log them at a higher serverity (don't want to be swallowing bad_allocs at the not-usually-logged debug severity).

Possibly also re-throw them, after resetting the reader, if that doesn't break anything.

Reply:
Thanks for the explanation. Makes sense.

Re-throwing should be fine here. If we don't throw here, the Kafka layer would throw upon inspecting the result of the read. I updated the PR.